dbms/src/Functions/FunctionHelpers.cpp

```diff
@@ -119,6 +119,75 @@ void validateArgumentType(const IFunction & func, const DataTypes & arguments,
                         ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
 }
 
+namespace
+{
+void validateArgumentsImpl(const IFunction & func,
+                           const ColumnsWithTypeAndName & arguments,
+                           size_t argument_offset,
+                           const FunctionArgumentTypeValidators & validators)
+{
+    for (size_t i = 0; i < validators.size(); ++i)
+    {
+        const auto argument_index = i + argument_offset;
+        if (argument_index >= arguments.size())
+        {
+            break;
+        }
+
+        const auto & arg = arguments[i + argument_offset];
+        const auto validator = validators[i];
+        if (validator.validator_func(*arg.type) == false)
```

Comment:
`!validator.validator_func(*arg.type)`

Reply:
fixed, even though I personally prefer `== false` here as more explicit and easier to catch by the naked eye.